### PR TITLE
Add method to retrieve URLs including artifacts URL

### PR DESCRIPTION
**Pull request by maelle**
cf #197, feature request by @noamross

Example. Should there be a method returning a character (vector) of the artifacts URLs directly? 

``` r
lala <- rhub::list_package_checks("/home/maelle/Documents/R-hub/test-packages/note",
                                  email = "maelle.salmon@yahoo.se",
                                  howmany = 4)

group_id <- lala$summary()$group_id[1]
group_check <- lala$get_group(group_id)
group_check$urls()
#>                                                                                    html
#> 1 https://builder-test.r-hub.io/status/note_1.0.tar.gz-4f9113c77256452791043ac07787ada8
#> 2 https://builder-test.r-hub.io/status/note_1.0.tar.gz-6761ed27a295401286be600994037c12
#> 3 https://builder-test.r-hub.io/status/note_1.0.tar.gz-f8bc9d2e812e42ba88dae2002c300d0b
#>                                                                                             text
#> 1 https://builder-test.r-hub.io/status/original/note_1.0.tar.gz-4f9113c77256452791043ac07787ada8
#> 2 https://builder-test.r-hub.io/status/original/note_1.0.tar.gz-6761ed27a295401286be600994037c12
#> 3 https://builder-test.r-hub.io/status/original/note_1.0.tar.gz-f8bc9d2e812e42ba88dae2002c300d0b
#>                                                                     artifacts
#> 1 https://artifacts.r-hub.io/note_1.0.tar.gz-4f9113c77256452791043ac07787ada8
#> 2 https://artifacts.r-hub.io/note_1.0.tar.gz-6761ed27a295401286be600994037c12
#> 3 https://artifacts.r-hub.io/note_1.0.tar.gz-f8bc9d2e812e42ba88dae2002c300d0b
group_check$urls()$artifacts
#> [1] "https://artifacts.r-hub.io/note_1.0.tar.gz-4f9113c77256452791043ac07787ada8"
#> [2] "https://artifacts.r-hub.io/note_1.0.tar.gz-6761ed27a295401286be600994037c12"
#> [3] "https://artifacts.r-hub.io/note_1.0.tar.gz-f8bc9d2e812e42ba88dae2002c300d0b"

check_id <- lala$summary()$check_id[4]
single_check <- lala$get_check(check_id)
single_check$urls()
#>                                                                                    html
#> 1 https://builder-test.r-hub.io/status/note_1.0.tar.gz-d3b8509d709e7ce30956ae764388887c
#>                                                                                             text
#> 1 https://builder-test.r-hub.io/status/original/note_1.0.tar.gz-d3b8509d709e7ce30956ae764388887c
#>                                                                     artifacts
#> 1 https://artifacts.r-hub.io/note_1.0.tar.gz-d3b8509d709e7ce30956ae764388887c
single_check$urls()$artifacts
#> [1] "https://artifacts.r-hub.io/note_1.0.tar.gz-d3b8509d709e7ce30956ae764388887c"
```

<sup>Created on 2019-03-07 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

### CODE REVIEW

File: R/check-class.R [hunk 1]
```diff
@@ -102,17 +108,44 @@ check_update <- function(self, private) {
...
+         call. = FALSE)
   }
-
-  urls <- paste0(sub("/api$", "/status/", baseurl()), ids)
-  lapply(urls, browseURL)
-  invisible(self)
-}
+  
+  return(ids)
+}
```
Comment (gaborcsardi) [inline R/check-class.R]: You need a newline at the end of the file.

File: R/check-class.R [hunk 2]
```diff
@@ -102,17 +108,44 @@ check_update <- function(self, private) {
...
+  
+  do.call("rbind",
+          lapply(ids, get_check_urls))
+}
+
+get_check_urls <- function(id){
+  data.frame(html = paste0(sub("/api$", "/status/", baseurl()), id),
+             text = paste0(sub("/api$", "/status/original/", baseurl()), id),
+             artifacts = paste0("https://artifacts.r-hub.io/", id),
+             stringsAsFactors = FALSE)
```
Comment (gaborcsardi) [inline R/check-class.R]: I think the `paste0()` calls can be vectorized, and there  is no need for `rbind()`.

Comment (maelle) [reply R/check-class.R]: :woman_facepalming: thanks, now fixed!

**Comment by maelle** [comment]
Note that the PR doesn't handle the fact that artifacts and logs are not there forever.

File: R/check-class.R (hunk 1 above)
Comment (maelle) [reply R/check-class.R]: :heavy_check_mark:

**Comment by gaborcsardi** [comment]
> Note that the PR doesn't handle the fact that artifacts and logs are not there forever.

Good point, maybe we can add a note about this.

**Comment by maelle** [comment]
A `message()`?

**Comment by gaborcsardi** [comment]
No, just a note in the docs, I think.

**Comment by maelle** [comment]
Ok, done, and I'll make it clear in the README/vignette when I show such features.

File: R/check-class.R [hunk 3]
```diff
@@ -102,17 +111,38 @@ check_update <- function(self, private) {
...
+check_urls <- function(self, private, which) {
+  
+  ids <- select_ids(which = which, self = self, 
+                    private = private)
+  
+  data.frame(html = paste0(sub("/api$", "/status/", baseurl()), ids),
+             text = paste0(sub("/api$", "/status/original/", baseurl()), 
+                           ids),
+             artifacts = paste0("https://artifacts.r-hub.io/", ids),
+             stringsAsFactors = FALSE)
```
Comment (gaborcsardi) [inline R/check-class.R]: Actually, one more thing. I think `group` and `id` should be also in the data frame, no? If it is already a data frame....

Comment (maelle) [reply R/check-class.R]: only group_id and check_id or also other stuff like platform?

Comment (gaborcsardi) [reply R/check-class.R]: Yeah. But then we could probably call it sg else, and and have separate methods  that only return the log url, the artifact url, etc.

Comment (gaborcsardi) [reply R/check-class.R]: Anyway, I think I'll merge this now, and will improve after....

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**